src/test/java/com/ncc/aif/ScalingTest.java

```diff
@@ -259,62 +327,114 @@ private void addEventOrRelation() {
     }
 
     private void addEvent() {
-        // Add an event
+        // Add an event and add it to a cluster
         Resource eventResource = makeEvent(model, getEventUri(), system);
+        Resource eventCluster = makeClusterWithPrototype(model, getClusterUri(), eventResource, system);
+
+        if (introduceChaos(NIST_RESTRICTION.FlatClusters)) {
+            Resource entityCluster = makeClusterWithPrototype(model, getClusterUri(), getRandomEntity(), "handle", system);
+            markAsPossibleClusterMember(model, eventCluster, entityCluster, .5, system);
+        }
 
         // Set the type
-        Resource typeResource = SeedlingOntology.Business_DeclareBankruptcy;
+        Resource typeResource = LDCOntology.Conflict_Attack_AirstrikeMissileStrike;
         Resource typeAssertion = markType(model, getAssertionUri(), eventResource, typeResource, system, 1.0);
 
         addJustificationAndPrivateData(typeAssertion);
 
         // Make two arguments
         Resource argument = markAsArgument(model, eventResource,
-                SeedlingOntology.Business_DeclareBankruptcy_Place,
+                LDCOntology.Conflict_Attack_AirstrikeMissileStrike_Place,
                 getRandomEntity(), system, 0.785, getAssertionUri());
-        addJustificationAndPrivateData(argument);
+        if (preserveValidity(NIST_RESTRICTION.RestrictJustification_Edges)) {
```

Comment:
Why do you apply some restrictions to only Entities/Events/Relations, but you apply this to both Events and Relations?

Reply:
My approach was to find a reasonable place or places to introduce chaos (invalidities) in the generated model and apply the restrictions when I could (if it didn't interfere with another restriction), with the goal to generate enough of those invalidities on average.  So don't bury a restriction only in an edge case.

Because there tend to be fewer events and relations than entities, I decided to apply that particular restriction to both events and relations-- to make sure that enough invalidities were generated to make it meaningful.  That's also why EverythingClustered and NameMaxLength can only be introduced for entities, because there are many more of them.

Not sure if this makes sense, but let me know if you think a given restriction should be added/removed from one place or another.  If you think one restriction is being generated too much or too little-- with all of the FREQ constants currently set to 0.3-- then let me know.